compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/DerivedConfigUtils.cpp

```diff
@@ -181,4 +181,31 @@ SmallVector<int64_t> deriveThreadTileSizes(Operation *op) {
       .Default([&](Operation *op) -> SmallVector<int64_t> { return {}; });
 }
 
+static const int64_t kDefaultGlobalLoadBitSizePerThread =
+    32; // TODO: make it a query.
+SmallVector<int64_t> globalLoadDMATileSizes(Operation *op) {
+  auto funcOp = op->getParentOfType<FunctionOpInterface>();
+
+  std::optional<SmallVector<int64_t>> workgroupSize = getWorkgroupSize(funcOp);
+  if (!workgroupSize) {
+    return {};
+  }
+  [[maybe_unused]] int64_t numThreads =
+      std::accumulate(workgroupSize->begin(), workgroupSize->end(), 1,
+                      std::multiplies<int64_t>());
+  assert(isa<linalg::LinalgOp>(op) && "expected linalg op");
+  auto linalgOp = cast<linalg::LinalgOp>(op);
```

Comment:
`dyn_cast` and `return {}` if it's not a linalg op. Asserts should only be for things that are logically impossible (but can be hit if a compiler dev writes buggy code, e.g. API misuse).